### ISSUE TITLE
An improvement is planned to fasten large calculations

### ISSUE THREAD

**Issue by paaksing** [paaksing/django-cassiopeia#1]
# Issue presented by maintainer:
After making some production tests, the overall flow is slowing down when doing large and extense calculations. I am aware of this issue since today, the reason it is slowing down is due to the fact that the Django-Cache is not a python object, so it needs to be `pickled` to store it and `unpickled` to pull it back. each time it pickles or unpickles it takes around 0.02 sec (in my own pc), _**but then doing extense calculations, e.g. when fetching all items+runes+spells+champs+everything in a match it normally have to pull some  300 times, then the 0.02 sec can sum up pretty quickly to around 6 entire seconds.**_
# Temporary Solution (for definitive solution read below):
A temporary solution in case you would need to do this __kind__ of calculations would be to add the standard `Cache` from the original `cassiopeia` (no need to install, it is a dependency) right before the `DjangoCache` *__with a LOW AMOUNT OF EXPIRATION to prevent memory issues, then at the end of your calculations make sure to flush the cache (a lot better if you do it at start too)__*. As follow:
* Change the `CASSIOPEIA_PIPELINE` in your settings.py
```python
CASSIOPEIA_PIPELINE = {
    # Include the cassiopeia.core objects that you see is constantly pulling
    # e.g. champions, runes, items, summoner spells
    # Do not cache datas that are pull once, e.g. Matches in the scenario above
    "Cache" : {
        "expirations_map" : {
            td(minutes=30): ["r", "r+", "i", "i+", "ss", "ss+", "c", "c+"],
            0: ["*+"]
        },
    },
    "DjangoCache" : {},
    "DDragon": {},
    "RiotAPI": {},
}
```
* Now in your extense calculated functions:
```python
from django_cassiopeia import cassiopeia as cass

def imhuge(args):
    cass.configuration.settings.expire_sinks()
    # flush at start
    # big calculation going here
    # flush at end
    cass.configuration.settings.expire_sinks()
```
Above is to make sure you don't run into memory issues and also get your performance good temporary
# Definitive Fix is scheduled on version 1.2.0
A fix to this is scheduled on next release, __expected realease time around August with the next big boy release: Django 3.1__**. The idea of this fix is to give a temporary python object Cache for each long term extensive calculation (which most likely done periodically), the mechanism used may be in a form similar to `@method_decorator` which you can pass the needed time to stay inside the object, and its memory usage is aborted after function execution.

*__Currently I am deciding between if this object is accessible globally during the execution of the main function, meaning that other functions that runs in pararel will benefit from it too, or just restrict to functions that has the decorator (which needs more work for me to do lol) and possibly role out a more perfect rate limiter.__*

**Comment by paaksing** [comment]
# Fixed on next deployment (1.3.0)
A new deployment is going live tomorrow (July 26th) in version 1.3, the implementation has changed, now you will get a brand new `datastore` called `Omnistone` which will have a built in cull system similar to Django's database backed cache -> LRU. It can cache all `core` objects, the cache is meant to be temporary and not long term so it will shut down if Django's server stops, YOU SHOULD NOT use it to cache static data like `Match` or `Summoner` which has millions of entries, a safe lock is put in place, you can bypass this lock but not recommended.

**Comment by paaksing** [comment]
# Fixed in 2.0

Imprementation was to add a new datastore capable to store `cassiopeia.core` objects that lives on server runtime, which is 5x faster than pulling from a cache server or database. More info is in the new [documentations](https://paaksing.github.io/django-cassiopeia/documentation/)
